### ISSUE TITLE
zio-test: setting env vars on TestAspect.beforeAll

### ISSUE THREAD

**Issue by mihaisoloi** [zio/zio#7775]
When using `TestAspect.beforeAll` env vars don't get set using `TestSystem.putEnv`. 

I understand that ZIO-test provides each test with a separate copy of the test environment, perhaps we can set the env vars in all the copies.

Minimum reproducible example:
https://scastie.scala-lang.org/K8OD7ZdUQQq4Qt6qLhFZyg

**Comment by mihaisoloi** [comment]
I will see if I can fix this

**Comment by adamgfraser** [comment]
I don't think this makes sense because we need to treat all of the test services consistently. With `TestEnvironment` we could copy the environment variables from the outer scope and create a new instance with these variables since they are just data. But with a service such as the `TestClock` it contains not just data but an actual queue of promises of work that needs to be done. Copying these to other instances would create interference between tests that we need to avoid. This use case can easily be addressed by using `before` instead of `beforeAll`.

**Comment by mihaisoloi** [comment]
I understand your point, but it gets a bit wasteful to use `before` when starting up test containers for example, and setting the environment parameters for the configs to read into the integration tests.

Can we at least add a clarification in the zio docs that this only works with `before` and not `beforeAll`? Here for example: https://zio.dev/reference/test/aspects/before-after-around

**Comment by adamgfraser** [comment]
I can definitely add a comment to the documentation about that!
